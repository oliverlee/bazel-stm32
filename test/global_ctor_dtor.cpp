#include <cstdio>

// NOLINTBEGIN(cppcoreguidelines-pro-type-vararg)

struct chatty
{
    ~chatty()
    {
        ::printf("~chatty()\n");
    }
    chatty()
    {
        ::printf("chatty()\n");
    }
};

const auto global = chatty{};

auto main() -> int
{
  ::printf("in main\n");
}

// NOLINTEND(cppcoreguidelines-pro-type-vararg)
