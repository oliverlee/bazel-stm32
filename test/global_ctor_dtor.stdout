QEMU 9.2.4 monitor - type 'help' for more information
(qemu) chatty()
in main
~chatty()
